src/client/runner/automation/playback/press/index.js

```diff
@@ -0,0 +1,152 @@
+    }
+
+    static _getShortcuts (keyCombination) {
+        var keys               = getKeyArray(keyCombination.toLowerCase());
+        var shortcuts          = [];
+        var curFullCombination = [];
+        var curCombination     = [];
+
+        for (var i = 0; i < keys.length; i++) {
+            curFullCombination.push(keys[i]);
+
+            curCombination = curFullCombination.slice();
+
+            while (curCombination.length) {
+                var keyString = curCombination.join('+');
+
+                if (supportedShortcutHandlers[keyString]) {
+                    shortcuts.push(keyString);
+                    curFullCombination = curCombination = [];
+                }
+                else
+                    curCombination.shift();
+            }
+        }
+
+        return shortcuts;
+    }
+
+    static _getShortcutHandlers (keyCombination) {
+        var shortcuts          = PressAutomation._getShortcuts(keyCombination.toLowerCase());
+        var shortcutHandlers   = {};
+        var stringWithShortcut = '';
+        var shortcut           = null;
+        var shortcutPosition   = null;
+        var shortcutLength     = null;
+
+        for (var i = 0; i < shortcuts.length; i++) {
+            shortcut         = shortcuts[i];
+            shortcutPosition = keyCombination.indexOf(shortcut);
+            shortcutLength   = shortcut.length;
+
+            stringWithShortcut += keyCombination.substring(0, shortcutPosition + shortcutLength);
+
+            shortcutHandlers[stringWithShortcut] = supportedShortcutHandlers[shortcut];
+
+            keyCombination = keyCombination.substring(shortcutPosition + shortcutLength);
+        }
+
+
+        return shortcutHandlers;
+    }
+
+    _down (keyPressSimulator) {
+        this.pressedKeyString += (this.pressedKeyString ? '+' : '') + keyPressSimulator.key;
+
+        var keyDownPrevented = !keyPressSimulator.down(this.modifiersState);
+
+        return Promise.resolve(keyDownPrevented);
+    }
+
+    _press (keyPressSimulator, keyEventPrevented) {
+        // NOTE: preventing the 'keydown' and 'keypress' events for the select element does not
+        // affect the assignment of the new selectedIndex. So, we should execute a shortcut
+        // for the select element without taking into account that 'key' events are suppressed
+        if (keyEventPrevented && !this.isSelectElement)
+            return wait(KEY_PRESS_DELAY);
+
+        var currentShortcutHandler = this.shortcutHandlers[this.pressedKeyString];
+        var keyPressPrevented      = false;
+
+        // NOTE: B254435
+        if (!currentShortcutHandler || browserUtils.isFirefox || keyPressSimulator.key === 'enter')
+            keyPressPrevented = !keyPressSimulator.press(this.modifiersState);
+
+        if ((!keyPressPrevented || this.isSelectElement) && currentShortcutHandler) {
+            return currentShortcutHandler(domUtils.getActiveElement())
+                .then(wait(KEY_PRESS_DELAY));
```

Comment:
Here and after: it's incorrect - `wait` will start executing with the `currentShortcutHandler`, so delay include time spent in `currentShortcutHandler`, while it should run after it.